boilerplate/bin/generate/component.mjs

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env zx
+import "zx/globals"
+import { prettier } from "../tools/string.mjs"
+import { dir } from "../tools/path.mjs"
+import { update } from "../tools/patch.mjs"
+
+const name = await question("Component name? ")
+const fileName = name;
+const ext = "tsx"
+
+const file = prettier(
+  /*ts*/ `
+import * as React from "react"
+import { StyleProp, TextStyle, View, ViewStyle } from "react-native"
+import { observer } from "mobx-react-lite"
+import { colors, typography } from "../theme"
+import { Text } from "./Text"
+
+export interface NameProps {
+  /**
+   * An optional style override useful for padding & margin.
+   */
+  style?: StyleProp<ViewStyle>
+}
+
+/**
+ * Describe your Name here
+ */
+export const Name = observer(function Name(props: NameProps) {
+  const { style } = props
+  const $styles = [$container, style]
+
+  return (
+    <View style={$styles}>
+      <Text style={$text}>Hello</Text>
+    </View>
+  )
+})
+
+const $container: ViewStyle = {
+  justifyContent: "center",
+}
+
+const $text: TextStyle = {
+  fontFamily: typography.primary.normal,
+  fontSize: 14,
+  color: colors.palette.primary500,
+}
+`,
+  { Name: name },
```

Comment:
Instead of using template literals for file strings, I decided to go with doing a regex replace. 

This was for a few reasons:
- using template literal syntax breaks [syntax highlighting from this extension](https://marketplace.visualstudio.com/items?itemName=zjcompt.es6-string-javascript)
- I find it's hard to read

A trade off is that it's not immediately obvious which strings are variables in a template string. But I still find it to be pretty readable, and one can always use Ctrl+F to see what is getting find and replaced